packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -613,40 +631,104 @@ export class TSRHandler {
 			}
 		}
 
-		await Promise.race([
-			Promise.all(ps),
-			new Promise<void>((resolve) =>
+		const resultsOrTimeout = await Promise.race([
+			Promise.all(ps).then(
+				(result) =>
+					({
+						success: true,
+						result,
+					} as UpdateDeviceOperationsResult)
+			),
+			new Promise<UpdateDeviceOperationsResult>((resolve) =>
 				setTimeout(() => {
 					const keys = _.keys(promiseOperations)
 					if (keys.length) {
 						this.logger.warn(`Timeout in _updateDevices: ${keys.join(',')}`)
 					}
-					resolve()
+					resolve({
+						success: false,
+						reason: 'timeout',
+						details: keys,
+					} as UpdateDeviceOperationsResult)
 				}, INIT_TIMEOUT)
 			), // Timeout if not all are resolved within INIT_TIMEOUT
 		])
-		ps = []
-
-		// Set logDebug on the devices:
-		for (const device of this.tsr.getDevices()) {
-			const deviceOptions = devices.get(device.deviceId)
-			if (deviceOptions) {
-				const debug: boolean = this.getDeviceDebug(deviceOptions)
-				if (device.debugLogging !== debug) {
-					this.logger.info(`Setting logDebug of device ${device.deviceId} to ${debug}`)
-					ps.push(device.setDebugLogging(debug))
+
+		await this._reportResult(resultsOrTimeout)
+
+		{
+			const ps: Promise<any>[] = []
+			// Set logDebug on the devices:
+			for (const device of this.tsr.getDevices()) {
+				const deviceOptions = devices.get(device.deviceId)
+				if (deviceOptions) {
+					const debug: boolean = this.getDeviceDebug(deviceOptions)
+					if (device.debugLogging !== debug) {
+						this.logger.info(`Setting logDebug of device ${device.deviceId} to ${debug}`)
+						ps.push(device.setDebugLogging(debug))
+					}
 				}
 			}
+			await Promise.all(ps)
 		}
-		await Promise.all(ps)
 
 		this._triggerupdateExpectedPlayoutItems() // So that any recently created devices will get all the ExpectedPlayoutItems
 		this.logger.info('updateDevices end')
 	}
 	private getDeviceDebug(deviceOptions: DeviceOptionsAny): boolean {
 		return deviceOptions.debug || this._coreHandler.logDebug || false
 	}
-	private async _addDevice(deviceId: string, options: DeviceOptionsAny): Promise<any> {
+	private async _reportResult(resultsOrTimeout: UpdateDeviceOperationsResult): Promise<void> {
```

Comment:
This method could perhaps use some comments in the code to clarify what it's doing at the different stages.

One of the things that confused me upon initial read was the difference between `resultsOrTimeout.success` (which actually means "timed out or not" - is that the intention?) and `result.success` ("device initalized successfully").